### PR TITLE
ci: CodeQL

### LINKED ISSUE #22: Use Github's standardized code scanning interface for Actions
Github provides [CodeQL](https://securitylab.github.com/tools/codeql) as a standardized interface to scanning code in the CI. This simplifies implementation and reporting. Several scanners are available on the [Marketplace](https://github.com/marketplace), e.g.:
- https://github.com/marketplace/actions/bandit-check
- https://github.com/marketplace/actions/aqua-security-trivy

This could be used to enhance the current CI and also add periodic scans.

### PR DESCRIPTION
**Pull request by Starkteetje**
Adds CodeQL pipeline checks for pushes and PRs to master and a regular cronjob checking the master branch

Fix #22

### CODE REVIEW

**Comment by Starkteetje** [comment]
Generated alert: https://github.com/sse-secure-systems/connaisseur/security/code-scanning/1?query=ref%3Arefs%2Fpull%2F51%2Fmerge

**Review by phbelitz — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '40 6 * * 3'
```
Comment (phbelitz) [inline .github/workflows/codeql-analysis.yml:9]: seems arbitrary, but sure

**Review by phbelitz — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (Starkteetje) [reply .github/workflows/codeql-analysis.yml:9]: Was Github's default when creating a CodeQL workflow :man_shrugging:

Comment (Starkteetje) [reply .github/workflows/codeql-analysis.yml:9]: So, yeah. Is arbitrary :P

**Review by phbelitz — APPROVED** [review]

**Pull request merged**